tests/functional/BaseCase.php

```diff
@@ -42,74 +40,176 @@
+
     protected GCPClientManager $clientManager;
 
-    // to distinguish projects if you need more projects in one test case
-    protected string $projectSuffix = '';
+    protected Log $log;
 
-    protected string $testRunId;
+    protected static function getRand(): string
+    {
+        return substr(md5(uniqid((string) mt_rand(), true)), 0, 3);
+    }
+
+    private function initialize(): void
+    {
+        if (file_exists('/tmp/prj0-cred')) {
+            unlink('/tmp/prj0-cred');
+        }
+        if (file_exists('/tmp/prj0-res')) {
+            unlink('/tmp/prj0-res');
+        }
+        if (file_exists('/tmp/prj0-id')) {
+            unlink('/tmp/prj0-id');
+        }
+
+        if (file_exists('/tmp/prj1-cred')) {
+            unlink('/tmp/prj1-cred');
+        }
+        if (file_exists('/tmp/prj1-res')) {
+            unlink('/tmp/prj1-res');
+        }
+        if (file_exists('/tmp/prj1-id')) {
+            unlink('/tmp/prj1-id');
+        }
+
+        $this->dropProjects($this->getStackPrefix());
+        $nameGenerator = new NameGenerator($this->getStackPrefix());
+        $suffix = date('mdHis') . self::getRand();
+        $project0 = [
+            ...$this->createProject('main-' . $suffix),
+            $nameGenerator->createProjectId('main-' . $suffix),
+        ];
+        $project1 = [
+            ...$this->createProject('link-' . $suffix),
+            $nameGenerator->createProjectId('link-' . $suffix),
+        ];
+        file_put_contents('/tmp/prj0-cred', $project0[0]->serializeToJsonString());
+        file_put_contents('/tmp/prj0-res', $project0[1]->serializeToJsonString());
+        file_put_contents('/tmp/prj0-id', $project0[2]);
+
+        file_put_contents('/tmp/prj1-cred', $project1[0]->serializeToJsonString());
+        file_put_contents('/tmp/prj1-res', $project1[1]->serializeToJsonString());
+        file_put_contents('/tmp/prj1-id', $project1[2]);
+    }
 
     protected function setUp(): void
     {
+        $this->log = new Log($this->getName(false));
+        $this->log->setPrefix($this->getTestHash());
+        $this->log->add($this->getName());
+        $GLOBALS['log'] = $this->log;
+        if (!file_exists('/tmp/initialized')) {
+            $store = new FlockStore('/tmp/test-lock');
+            $factory = new LockFactory($store);
+            $lock = $factory->createLock('init');
+            $this->log->add('init lock');
+            if ($lock->acquire(true)) {
+                if (!file_exists('/tmp/initialized')) {
+                    $this->log->add('initialize');
+                    $this->initialize();
+                    touch('/tmp/initialized');
+                    $lock->release();
+                } else {
+                    $this->log->add('initialized');
+                }
+            } else {
+                $this->log->add('not locked');
+            }
+        } else {
+            $this->log->add('already initialized');
+        }
```

Comment:
Je potřeba aby sa vytvořil na žáčtku paratestů jeden projekt, v dokumentaci paratestu je to dost na :hankey: tvrdí tam, že to ide udělat přes statiku, ale to teda nefungovalo, ani nevím jak by to teda fungovalo když je to každé zvlášť thread.
Takže sem tam dal lock na soubor a všecko co si potřebuju předat mezi testama (především údaje o projektoch) sem naházal taky do souboru.

Takže je to těď tak, že sa spustí x threadů paratestu, jeden vyhraje a vytvoří projekt, ostatní čekajů na zámek. A je projekt vytvořený, tak sa údaje uložíja do souboru a odemnkne sa zámek, pak si každý thread može načíst údaje pro běh.